src/twisted/newsfragments/10273.bugfix

```diff
@@ -0,0 +1 @@
+Deprecated ``threading.currentThread()`` in favor of ``threading.current_thread()`` to avoid the deprecation warnings introduced for the former in Python 3.10.
```

Comment:
If you keep the deprecation please use its fully-qualified name (i.e. `twisted.python.threadpool.ThreadPool.currentThread()`) in this news fragment.